temporian/core/data/event.py

```diff
@@ -97,7 +98,7 @@ def set_creator(self, creator):
 
 def input_event(
     features: List[Feature],
-    index: List[str] = [],
+    index: Dict[str, IndexDtypes] = {},
```

Comment:
As per today's meeting, if this is too hard to do in this PR, can you add a todo to transform this into a List?